rc/core/c-family.kak

```diff
@@ -143,12 +143,13 @@ evaluate-commands %sh{
             add-highlighter shared/FT/string region %{MAYBEAT(?<!QUOTE)(?<!QUOTE\\)"} %{(?<!\\)(?:\\\\)*"} fill string
             add-highlighter shared/FT/raw_string region %{R"([^(]*)\(} %{\)([^")]*)"} fill string
             add-highlighter shared/FT/comment region /\* \*/ fill comment
-            add-highlighter shared/FT/line_comment region // $ fill comment
+            add-highlighter shared/FT/line_comment region // (?<!\\)(?=\n) fill comment
             add-highlighter shared/FT/disabled region -recurse "#\h*if(?:def)?" ^\h*?#\h*if\h+(?:0|FALSE)\b "#\h*(?:else|elif|endif)" fill rgb:666666
-            add-highlighter shared/FT/macro region %{^\h*?\K#} %{(?<!\\)\n} group
+            add-highlighter shared/FT/macro region %{^\h*?\K#} %{(?<!\\)(?=\n)} group
 
             add-highlighter shared/FT/macro/ fill meta
             add-highlighter shared/FT/macro/ regex ^\h*#include\h+(\S*) 1:module
+            add-highlighter shared/FT/macro/ regex /\*.*?\*/ 0:comment
```

Comment:
I don't think comments should be highlighted within macros, as they are supposed to be interpreted by the compiler, whereas macros are a pre-processor thing (two separate layers). It's more consistent to highlight the entire macro as such, and be agnostic of its content.

Reply:
macroses can be huge, I once saw a macro that is 87 line long. It was something like implementation of template for plain C. It was filled with meaningful comments, explaining some complex preprocessor tricks. It is just easier to read such macro, when comments are comments.

Reply:
I understand, but isn't it common practice in most popular editors? In Vim and Neovim for sure comments are highlighted as comments in defines.

Reply:
After some experimenting, I found that the order is `line joining -> comment stripping -> macro processing` so comments preempt macros, which i think should be reflected in the highlighting. I think it would be best to show as closely as possible how the compiler is gonna read the code.